fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -304,7 +343,7 @@ public void deleteFile(@Nonnull String name) throws IOException {
     @Override
     public long fileLength(@Nonnull String name) throws NoSuchFileException {
         LOGGER.trace("fileLength -> {}", name);
-        FDBLuceneFileReference reference = getFDBLuceneFileReference(name).join();
+        FDBLuceneFileReference reference = context.instrument(FDBStoreTimer.Events.LUCENE_FILE_LENGTH,getFDBLuceneFileReference(name)).join();
```

Comment:
As mentioned above, this should probably be `asyncToSync()` and be a `Wait` instead of an `Event`